#!/bin/sh -e
## 03_zero-headers.dpatch by Jeff Johnson <jbj@redhat.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure created database header is initialized (RH #4457).

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/james/debian/packages/gdbm173/gdbm173-1.7.3/gdbmopen.c gdbm173-1.7.3/gdbmopen.c
--- /home/james/debian/packages/gdbm173/gdbm173-1.7.3/gdbmopen.c	1993-12-26 12:57:32.000000000 +0000
+++ gdbm173-1.7.3/gdbmopen.c	2003-05-05 17:51:26.000000000 +0100
@@ -208,7 +208,7 @@
 	file_block_size = block_size;
 
       /* Get space for the file header. */
-      dbf->header = (gdbm_file_header *) malloc (file_block_size);
+      dbf->header = (gdbm_file_header *) calloc (1, file_block_size);
       if (dbf->header == NULL)
 	{
 	  gdbm_close (dbf);
